### PR TITLE
feat: Added optional isAnimated prop to illustrations that support an…

### PR DESCRIPTION
**Pull request by ActuallyACat**
# Objective
Add an optional isAnimated property to the mood spot illustrations. There's a few things going on in this change: 

* **The same Spot component used for illustrations is used for animations.** This means if you would like to animate an existing mood spot (cautionary, positive, informative, etc) a user would just need to add an `isAnimated` prop to that component. 
* We fallback to a static image if the request for the lottie animation failed, or there was an error derserializing the lottie file (i.e if it's corrupt)

**Note that this is a breaking change**. Consumers will need to update their mocks in tests. 

Story: https://dev.cultureamp.design/ally/animated-illustrations/storybook/?path=/story/illustration-spot-react--animated-spot

# Checklist

- [x] If this contains visual changes, has it been reviewed by a designer?
- [x] If this introduces either a new component or a breaking change to an existing component, has it been reviewed by an advocate? (Ask in #prod_design_systems)
- [x] I have considered likely risks of these changes and got someone else to QA as appropriate
- [ ] I have or will communicate these changes to the front end practice

# Additional Considerations
- Does there need to be right-to-left (RTL) options for localization and internationalization?
- Has the test suite been updated?
- Have you done cross-browser testing for our [supported browsers](https://academy.cultureamp.com/hc/en-us/articles/204539569-Supported-browsers-for-Participants)?
- Have you updated any relevant documentation or left useful comments in the code?
- Have you reviewed Culture Amp's Web Accessibility guide [Current Compliance, Going Forward, Statement and Answers for Customers](https://cultureamp.atlassian.net/wiki/spaces/Prod/pages/428572998/Web+Accessibility)?
- If this contains substantial visual changes, especially far reaching ones, have you unlocked the Chromatic step in the pipeline?
- Have Storybook stories been updated?

**If you're new to Kaizen, please ask #prod_design_systems to set up an onboarding session to get you up to speed.** If you have an urgent PR to merge before that happens, it is safest to ask in #prod_design_systems to have a design system advocate review the PR to catch any issues.

### CODE REVIEW

File: draft-packages/illustration/KaizenDraft/Illustration/AnimatedBase.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { assetUrl } from "@kaizen/hosted-assets"
+import { BaseProps } from "./Base"
+import { getAnimationData } from "./utils"
+import styles from "./style.module.scss"
+import { LottieAnimation } from "./types"
+
+export type AnimatedBaseProps = {
+  autoplay?: boolean
+  loop?: boolean
+  isAnimated?: boolean
```
Comment (dougmacknz) [inline draft-packages/illustration/KaizenDraft/Illustration/AnimatedBase.tsx:12]: Do you actually need `isAnimated` here?

**Review by dougmacknz — CHANGES_REQUESTED** [review]
Awesome to see these animated! I found a few things that should probably be addressed, otherwise it's looking good to me.

File: draft-packages/illustration/KaizenDraft/Illustration/AnimatedBase.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  const LoadingState = (
+    /* Avoid jump when asset loads */
+    <div className={wrapper}>
+      <svg
+        data-testid="loading"
+        viewBox="0 0 100 100"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <circle fill="#f9f9f9" cx="48" cy="48" r="48" />
```
[the remark this answers is not in the document]
Comment (ActuallyACat) [reply draft-packages/illustration/KaizenDraft/Illustration/AnimatedBase.tsx:83]: Thoughts on what to include in the loading title? I'm going to hazard a guess in _most_ scenarios the illustration is purely decorative and we won't want to add annotations to them/leave an empty alt on them. 

I can see maybe two options and I'm undecided on the best approach: 
1. Pass-through `alt` to failed/success/loading state elements. So if our users supplied `alt="Successfully submitted survey"` each of those element states would display "Successfully submitted survey". Similarly for empty alts. I'm unsure if a screen reader would re-announce those states.
2. Wrap the whole thing in a `figure` and have one `figcaption` for this component, and completely ignore the individual img/svg within this component (maybe add `aria-hidden="true"` to the loading element and inline svg)

[the remark this answers is not in the document]
Comment (dougmacknz) [reply draft-packages/illustration/KaizenDraft/Illustration/AnimatedBase.tsx]: Yeah I like option 2 as well 👍

File: draft-packages/illustration/KaizenDraft/Illustration/Spot.tsx [hunk 3]
```diff
@@ -14,78 +16,144 @@ const noZenIllustrationWarning = (illustrationName: string) => {
...
+  ...otherProps
+}: AnimatedSpotProps) => {
   const theme = useTheme()
   const illustrationPath =
     theme.themeKey === "zen"
       ? "illustrations/spot/moods-cautionary.svg"
       : "illustrations/heart/spot/moods-cautionary.svg"
 
-  return <Base {...props} name={illustrationPath} />
+  if (isAnimated) {
```
[the remark this answers is not in the document]
Comment (ActuallyACat) [reply draft-packages/illustration/KaizenDraft/Illustration/Spot.tsx:45]: I've added a warning for those cases, and fallback to a static asset.

**Review by dougmacknz — DISMISSED** [review]
awesome awesome awesome

**Comment by makenosound** [comment]
We don’t necessarily need to tackle it straight away, but I wonder if we should have `prefers-reduced-motion` fallbacks for these: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

**Comment by ActuallyACat** [comment]
> We don’t necessarily need to tackle it straight away, but I wonder if we should have prefers-reduced-motion fallbacks for these: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

Ah! Yes I completely overlooked this - I'll add those now and fallback to static images. Seems straightforward enough to squeeze in now.

**Comment by dougmacknz** [comment]
>We don’t necessarily need to tackle it straight away, but I wonder if we should have prefers-reduced-motion fallbacks for these

I considered this but came to a conclusion that these were _probably_ low movement enough, though if it were looping it would probably cross that threshold. If we leave them in I could check with Intopia if we want.

**Comment by ActuallyACat** [comment]
> I considered this but came to a conclusion that these were probably low movement enough, though if it were looping it would probably cross that threshold. If we want, we could leave them in and I can check with Intopia.

I'll hold off for now in that case. I'm curious to hear what their thoughts on the movement threshold would be. Scene illustrations would definitely pass that threshold.

**Review by dlimiter — APPROVED** [review]

**Pull request merged**